### ISSUE TITLE
Sensor values totally unreliable

### ISSUE THREAD

**Issue by chjohans** [iJacks1980/FanControl.CommanderPRO#19]
Just tried this plugin, I've been using FanControl for quite some time and I also have the HWiNFO plugin installed.

I have a Commander Pro with two temperature sensors, and at the moment no fans. They temperature sensors are detected just fine by HWiNFO, and show correct temperatures.

But the sensor values reported by this plugin is dead wrong. They change every second, and I've seen values from 0 to as high as 117. It can show 0 at and then jump to 117 the next second. Not at all reliable, and totally wrong of course.

Just let me know what you nede to debug this and I'll help as best as I can. Thanks.

**Comment by iJacks1980** [comment]
You cannot use this plugin and ANY other software that reads from the Corsair devices - this is down to how Corsair designed the protocol.  It's one or the other I'm afraid.

There is a workaround for HWiNFO64, go into the main setting for HWiNFO64 under the safety tab and uncheck the "CorsairLink and Asetek Support". This will prevent HWiNFO64 from trying to read from Corair Commander devices (fan, pump and temperature data)

**Comment by chjohans** [comment]
I am aware of that, and I had that option disabled in UwiNFO, but I had the "Corsair Service" running at the time and that was probably the culprit.-

But, it's indeed possible to run more than one monitoring programs that read Corsair sensors. See here:
http://rh-software.com/

Scroll down to "SMBus locking", I know for a fact that at least HWiNFO, SIV (from the site above), and Argus Monitor use this mutex lock mechanism to only allow one of them to access Corsair sensors at any given time (although Corsair is not specifically mentioned on that page). There is an email address to write to with the info that will give you all the info you will need to implement this.

Please consider this so that FanControl will "play nice" with other popular and frequently used monitoring software. :)

**Comment by iJacks1980** [comment]
Thanks or the suggestion, I'm looking into it, but it will require a rather significant rewrite (which I was planning anyway to streamline other aspects) to fully support so it won't be right away ;)

I've got a very very busy week (last week at current job) and spending next week away from home, so I probably won't have an update till next weekend at the very earliest.

Sadly, it still won't play nice with iCUE :(

**Comment by chjohans** [comment]
Yes, but it's the other way around, it's iCUE that doesn't play nice with anything :(

But thanks for considering this, I think the "target group" for something like this is more than likely to use some of the tools this will "play nice with" if you do implement the mutex locking. :)
